### PR TITLE
【Hackathon 6th No.43】add paddle tril_triu

### PR DESCRIPTION
**Pull request by Asthestarsfalll**
### Details:
add paddle tril and triu

### Reference
Code: https://github.com/PaddlePaddle/Paddle/blob/release/2.6/python/paddle/tensor/creation.py#L1431
Support data type: int32,  int64, float32, float64, float16, complex64, complex128 (openvino do not support float64, float16, complex64 complex128) 

### Python simulation code
```python
import paddle

a = paddle.randn((3, 4))


def func(x, diagonal=0, mode='triu'):
    h, w = x.shape
    horizontal_range = paddle.arange(0, w)
    horizontal_range = paddle.unsqueeze(horizontal_range, 0)
    vertical_range = paddle.arange(0, h)
    vertical_range = paddle.unsqueeze(vertical_range, 1)
    if diagonal != 0:
        vertical_range += diagonal
    if mode == 'triu':
        mask = horizontal_range >= vertical_range
    else:
        mask = horizontal_range <= vertical_range
    print(mask)
    return paddle.where(mask, x, paddle.zeros_like(x))


def tril(x, diagonal=0):
    return func(x, diagonal, 'tril')

def triu(x, diagonal=0):
    return func(x, diagonal, 'triu')



b = paddle.tril(a, -1)
c = tril(a, -1)
print(b)
print(c)
print((b==c).mean())

```

### Test

240411_19h35m47s_screenshot

waiting for:
- https://github.com/openvinotoolkit/openvino/pull/23010

### CODE REVIEW

**Comment by Asthestarsfalll** [comment]
@meiyang-intel Could you please take a look?

File: src/frontends/paddle/tests/test_models/gen_scripts/generate_tril_triu.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+def main():
+    data = np.random.randn(4, 5) * 10
+    triu("triu", data)
+    triu("triu_1", data, 1)
+    triu("triu_2", data, -1)
+    triu("triu_3", data, 10)
+    triu("triu_4", data, -3)
+    triu("triu_int32", data, dtype="int32")
+    triu("triu_int64", data, dtype="int64")
```
Comment (meiyang-intel) [inline src/frontends/paddle/tests/test_models/gen_scripts/generate_tril_triu.py:73]: openvino can support float16. Can it pass with type float16?

File: src/frontends/paddle/tests/test_models/gen_scripts/generate_tril_triu.py
Comment (meiyang-intel) [inline src/frontends/paddle/tests/test_models/gen_scripts/generate_tril_triu.py:1]: Can the test pass in paddle version 2.4, 2.5?

File: src/frontends/paddle/tests/test_models/gen_scripts/generate_tril_triu.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        cpu = paddle.static.cpu_places(1)
+        exe = paddle.static.Executor(cpu[0])
+        exe.run(paddle.static.default_startup_program())
+
+        outs = exe.run(feed={"x": x}, fetch_list=[triu_outs])
+
+        saveModel(
+            name,
+            exe,
+            feedkeys=["x"],
```
Comment (meiyang-intel) [inline src/frontends/paddle/tests/test_models/gen_scripts/generate_tril_triu.py:32]: There is CI build failure here.

File: src/frontends/paddle/tests/test_models/gen_scripts/generate_tril_triu.py (hunk 1 above)
Comment (Asthestarsfalll) [reply src/frontends/paddle/tests/test_models/gen_scripts/generate_tril_triu.py:73]: paddle tril or triu only support float16 on gpu

File: src/frontends/paddle/tests/test_models/gen_scripts/generate_tril_triu.py
Comment (Asthestarsfalll) [reply src/frontends/paddle/tests/test_models/gen_scripts/generate_tril_triu.py:1]: yes, I have tested.

File: src/frontends/paddle/tests/test_models/gen_scripts/generate_tril_triu.py (hunk 2 above)
Comment (Asthestarsfalll) [reply src/frontends/paddle/tests/test_models/gen_scripts/generate_tril_triu.py]: fixed

**Review by meiyang-intel — APPROVED** [review]

**Comment by meiyang-intel** [comment]
@ilya-lavrenov , could you help build jenkins?

**Comment by ilya-lavrenov** [comment]
build_jenkins

**Pull request merged**